chainstate/src/detail/mod.rs

```diff
@@ -688,5 +806,59 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
     }
 }
 
+// FIXME: this enum looks too specific to move it to error.rs, but this file is also not
```

Comment:
I'm OK with this going into error.rs. But feel free to create a module with better abstraction (not just moving things around, like I explained earlier).

Reply:
I don't have a better abstraction at this moment, but I don't really like the idea of moving this enum to `error.rs`. The reason is that this error type is "ad-hoc", it's supposed to be decomposed right after calling the function that returns it. So I don't want it to be wrapped by some other error type and propagated further.
Is it Ok to leave it here then?

Reply:
I've removed this FIXME for now.